ooniapi/services/ooniprobe/src/ooniprobe/routers/v1/probe_services.py

```diff
@@ -590,3 +589,46 @@ def random_web_test_helpers(th_list: List[str]) -> List[Dict]:
     for th_addr in th_list:
         out.append({"address": th_addr, "type": "https"})
     return out
+
+
+class GeoLookupResponse(BaseModel):
+    """
+    v:
+      type: integer
+      description: response format version
+    geolocation:
+      type: object
+      description: dict of ip addresses to dict of country code, ASN, AS Name strings
+    """
+    v: int
+    geolocation: Dict[str, Dict[str, str]]
+
+
+@router.post("/geolookup", tags=["ooniprobe"])
+async def geolookup(
+        request: Request,
+        response: Response,
+        cc_reader: CCReaderDep,
+        asn_reader: ASNReaderDep,
+) -> GeoLookupResponse:
+
+    # initial values probe_geoip compares with
+    probe_cc = "ZZ"
+    asn = "AS0"
+
+    # extract addresses from request
+    json_body = await request.json()
```

Comment:
You should create a request model to avoid parsing the json fields on your own, fast api alrady uses reflection to run many validations using type annotations. example:

https://github.com/ooni/backend/blob/5763024ff0859fa335971080b99f6ae89766160f/ooniapi/services/ooniprobe/src/ooniprobe/routers/reports.py#L105